### PR TITLE
John conroy/cell expression histogram

### PR DESCRIPTION
**Pull request by john-conroy**
Adds a very basic histogram component to shared styles and a histogram of gene expression for cells in a dataset to the cells UI. Moves diagnostic info into the histogram component for simplicity.

### CODE REVIEW

File: context/app/static/js/pages/CellsAPIDemo/CellExpressionHistogram/CellExpressionHistogram.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        margin={{
+          top: 50,
+          right: 50,
+          left: 50,
+          bottom: 50,
+        }}
+      />
+    </>
+  ) : (
+    <Typography>Please wait for histogram...</Typography>
```
Comment (mccalluc) [inline context/app/static/js/pages/CellsAPIDemo/CellExpressionHistogram/CellExpressionHistogram.jsx:46]: Maybe this is a place holder, but down the road, we probably want either a spinner, or a grey-box ghost?

File: context/app/static/js/pages/CellsAPIDemo/DatasetClusterChart/DatasetClusterChart.jsx [hunk 2]
```diff
@@ -98,7 +86,7 @@ function DatasetClusterChart({ uuid, geneName, minGeneExpression }) {
       />
     </>
   ) : (
-    <Typography>Please wait...</Typography>
+    <Typography>Please wait for cluster chart...</Typography>
```
Comment (mccalluc) [inline context/app/static/js/pages/CellsAPIDemo/DatasetClusterChart/DatasetClusterChart.jsx:89]: ditto.

File: context/app/static/js/shared-styles/charts/Histogram/Histogram.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  const yScale = useMemo(
+    () =>
+      scaleLinear({
+        domain: [0, Math.max(...chartData.map((d) => d.length))],
+        nice: true,
+        round: true,
+        range: [yHeight, 0],
+      }),
+    [chartData, yHeight],
+  );
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/charts/Histogram/Histogram.jsx:34]: Were you running into problems without `useMemo`? or is scaleLinear expensive? A little leery of adding complexity unless the benefits are clear... If it stays, maybe add a note above?

File: context/app/static/js/shared-styles/charts/Histogram/Histogram.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    <div>
+      <svg width={parentWidth} height={parentHeight}>
+        <GridRows top={margin.top} left={margin.left} scale={yScale} width={xWidth} stroke="black" opacity={0.38} />
+        <Group top={margin.top} left={margin.left}>
+          {chartData.map((d) => {
+            const barWidth = Math.max(0, xScale(d.x1) - xScale(d.x0) - 1);
+            const barHeight = yScale(0) - yScale(d.length);
+            const barX = xScale(d.x0) + 1;
+            const barY = yScale(d.length);
+            return <Bar x={barX} y={barY} width={barWidth} height={barHeight} fill="#6C8938" />;
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/charts/Histogram/Histogram.jsx:46]: pull color from theme?

**Review by mccalluc — COMMENTED** [review]
Looks good. Small comments. If `memo` stays, add explanation?

File: context/app/static/js/pages/CellsAPIDemo/CellExpressionHistogram/CellExpressionHistogram.jsx (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/pages/CellsAPIDemo/CellExpressionHistogram/CellExpressionHistogram.jsx:46]: Yes, that will come down the road.

**Pull request merged**